code/game/machinery/anomaly_pad.dm

```diff
@@ -0,0 +1,125 @@
+
+	if(L)
+		active_power_usage = base_power / L.rating
+		idle_power_usage = active_power_usage * 0.1
+
+/obj/machinery/anomaly_pad/attackby(obj/item/I, mob/living/user, params)
+
+	if(default_deconstruction_screwdriver(user, icon_state, icon_state, I))
+		update_icon_state()
+		return
+
+	else if(default_deconstruction_crowbar(I))
+		return
+
+	if(!anomaly && istype(I, /obj/item/assembly/signaler/anomaly) && user.dropItemToGround(I) && IsAnomalyApplicable(I, user))
+		CaptureAnomaly(I)
+		playsound(loc, 'sound/machines/click.ogg', 15, TRUE)
+		return
+
+	. = ..()
+
+/obj/machinery/anomaly_pad/process()
+	if(anomaly && (!is_operational() || get_turf(src) != get_turf(anomaly)))
+		ReleaseAnomaly()
+		return FALSE //uuhhh dont know if its ok but im using this to see if the check passed on the subtype
+
+	return TRUE
+
+///Spawn an anomaly from an anomaly core and suspend it ''''''''''safely''''''''''' on us
+/obj/machinery/anomaly_pad/proc/CaptureAnomaly(obj/item/assembly/signaler/anomaly/S)
+	anomaly = S.ReviveAnomaly(get_turf(src))
+	anomaly.Suspend()
+	use_power = ACTIVE_POWER_USE
+
+///let them loose into the world to reign havoc once more
+/obj/machinery/anomaly_pad/proc/ReleaseAnomaly()
+	anomaly.Unsuspend()
+	anomaly = null
+	playsound(src, 'sound/effects/supermatter.ogg', 50, TRUE)
+	use_power = IDLE_POWER_USE
+
+///Check if an anomaly is applicable. Example is the chemical pad only accepting fluescent anomalies
+/obj/machinery/anomaly_pad/proc/IsAnomalyApplicable(obj/item/assembly/signaler/anomaly/S, mob/living/user)
+	return TRUE
+
+/obj/machinery/anomaly_pad/Destroy()
+	if(anomaly)
+		ReleaseAnomaly()
+	return ..()
+
+/obj/machinery/anomaly_pad/update_icon()
+	. = ..()
+
+	icon_state = initial(icon_state) + (!powered(EQUIP) ? "-off" : "")
+
+/obj/machinery/anomaly_pad/update_overlays()
+	. = ..()
+	if(panel_open)
+		. += initial(icon_state) + "-panel"
+
+/obj/machinery/anomaly_pad/power_change()
+	. = ..()
+
+	update_icon_state()
+
+///Exactly the same as a normal anomaly pad, but this one only takes fluescent anomalies and can also drain their reagents.
+/obj/machinery/anomaly_pad/liquid
+	name = "chemical anomaly pad"
+	desc = "Drains a fluescent anomolies imprinted reagent with extreme efficiency."
+
+	icon_state = "pad" //unique icon
+
+	circuit = /obj/item/circuitboard/machine/anomaly_pad_liquid
+
+	///Max volume of the pad
+	var/volume = 1000
+	///Max anomaly drainage rate
+	var/speed = 100
+	///alright, it'd be gross to make another anomaly var for the fluescent anom, so just track the reagent_type here
+	var/reagent_type 
+
+/obj/machinery/anomaly_pad/liquid/Initialize()
+	. = ..()
+	create_reagents(volume)
+	AddComponent(/datum/component/plumbing/simple_supply, anchored)
+
+/obj/machinery/anomaly_pad/liquid/process()
+	. = ..()
+	if(!. || !anomaly)
+		return
+
+	reagents.add_reagent(reagent_type, speed)
+
+/obj/machinery/anomaly_pad/liquid/CaptureAnomaly(obj/item/assembly/signaler/anomaly/fluid/S)
+	..(S)
```

Comment:
```suggestion
	. = ..()
```
No need to pass the `S` to the parent, that will already be done by default.

Reply:
Also doesnt expect a return value, but I'll remove the S